generic/lshw.py

```diff
@@ -0,0 +1,177 @@
+        for package in ("lshw", "net-tools", "iproute", "pciutils"):
+            if not soft.check_installed(package) and not soft.install(package):
+                self.error("Fail to install %s required for this"
+                           " test." % package)
+
+    def test_lshw(self):
+        """
+        lshw without any options would generate full information
+        report about all detected hardware.
+        """
+        self.log.info("===============Executing lshw test ===="
+                      "===========")
+        self.run_cmd("lshw")
+        self.run_cmd("lshw -version")
+        if self.is_fail >= 1:
+            self.fail("%s command(s) failed to execute  "
+                      "verification" % self.is_fail)
+
+    def test_lshw_short(self):
+        """
+        With "-short" option,the lshw command would generate a brief
+        information report about the hardware devices
+        """
+        self.log.info("===============Executing lshw -short tests ===="
+                      "===========")
+        self.run_cmd("lshw -short")
+        self.run_cmd("lshw -short  -class network")
+        self.run_cmd("lshw -short  -class storage")
+        self.run_cmd("lshw -short  -class memory")
+        self.run_cmd("lshw -short  -class power")
+        self.run_cmd("lshw -short  -class bus")
+        self.run_cmd("lshw -short  -class processor")
+        self.run_cmd("lshw -short  -class system")
+        if self.is_fail >= 1:
+            self.fail("%s command(s) failed to execute "
+                      "verification" % self.is_fail)
+
+    def test_lshw_class(self):
+        """
+        To display information about any particular hardware,specify the class.
+        """
+        self.log.info("===============Executing lshw -class tests ===="
+                      "===========")
+        self.run_cmd("lshw -class disk -class storage")
+        self.run_cmd("lshw -class memory")
+        self.run_cmd("lshw -class cpu")
+        self.run_cmd("lshw -class volume")
+        self.run_cmd("lshw -class network")
+        self.run_cmd("lshw -class power")
+        self.run_cmd("lshw -class generic")
+        self.run_cmd("lshw -class processor")
+        self.run_cmd("lshw -class bridge")
+        self.run_cmd("lshw -class multimedia")
+        self.run_cmd("lshw -class display")
+        self.run_cmd("lshw -class system")
+        self.run_cmd("lshw -class communication")
+        self.run_cmd("lshw -class bus")
+        if self.is_fail >= 1:
+            self.fail("%s command(s) failed to execute "
+                      "verification" % self.is_fail)
+
+    def test_lshw_verification(self):
+        """
+        compare the output of lshw with other tools
+        which produces similar info of hardware.
+        """
+        # verifying mac address
+        get_mac = process.system_output(
+            " ip link show %s " % self.active_interface)
+        mac = re.search(r'link\/\ether (.*) brd', get_mac).group(1)
```

Comment:
This can be added to avocado utils, since others could use it as well. Please consider